pkg/prometheus/resource_selector_test.go

```diff
@@ -4588,6 +4588,231 @@ func TestSelectScrapeConfigs(t *testing.T) {
 			valid:       true,
 			scrapeClass: ptr.To("existent"),
 		},
+		{
+			scenario: "StackitSDconfig with supported prometheus version",
+			updateSpec: func(sc *monitoringv1alpha1.ScrapeConfigSpec) {
+				sc.StackitSDConfigs = []monitoringv1alpha1.StackitSDConfig{
+					{
+						Project: "11111111-1111-1111-1111-111111111111",
+						Region:  ptr.To("eu01"),
+						Authorization: monitoringv1.SafeAuthorization{
+							Credentials: &v1.SecretKeySelector{
+								LocalObjectReference: v1.LocalObjectReference{
+									Name: "secret",
+								},
+								Key: "credential",
+							},
+						},
+					},
+				}
+			},
+			valid:       true,
+			promVersion: "3.5.0",
+		},
+		{
+			scenario: "StackitSDconfig with unsupported prometheus version",
+			updateSpec: func(sc *monitoringv1alpha1.ScrapeConfigSpec) {
+				sc.StackitSDConfigs = []monitoringv1alpha1.StackitSDConfig{
+					{
+						Project: "11111111-1111-1111-1111-111111111111",
+						Region:  ptr.To("eu01"),
+						Authorization: monitoringv1.SafeAuthorization{
+							Credentials: &v1.SecretKeySelector{
+								LocalObjectReference: v1.LocalObjectReference{
+									Name: "secret",
+								},
+								Key: "credential",
+							},
+						},
+					},
+				}
+			},
+			valid:       false,
+			promVersion: "3.2.0",
+		},
+		{
+			scenario: "StackitSDconfig with invalid TLS config with invalid CA data",
+			updateSpec: func(sc *monitoringv1alpha1.ScrapeConfigSpec) {
+				sc.StackitSDConfigs = []monitoringv1alpha1.StackitSDConfig{
+					{
+						Project: "11111111-1111-1111-1111-111111111111",
+						Region:  ptr.To("eu01"),
+						TLSConfig: &monitoringv1.SafeTLSConfig{
+							CA: monitoringv1.SecretOrConfigMap{
+								Secret: &v1.SecretKeySelector{
+									Key: "invalid_ca",
+									LocalObjectReference: v1.LocalObjectReference{
+										Name: "secret",
+									},
+								},
+							},
+						},
+					},
+				}
+			},
+			valid:       false,
+			promVersion: "3.2.0",
```

Comment:
```suggestion
			promVersion: "3.5.0",
```

I would rather keep the supported version just to make sure the test fails due to invalid CA data